### DISCUSSION TITLE
Best way to handle 404s in a defined route

### DISCUSSION THREAD

**Discussion by studiopatagon** [jeremyevans/roda#313] [General]
I have the not_found plugin installed which works great for non-defined routes. But if I have a route that is defined such as

/news/:id

But then the :id is non-existent in the database, what's the best way to turn it into a 404 and keeping it DRY?

Currently I'm doing this, which just shows a 404 view, but does not have the correct response header:

```
      if @content
          view :single
      else
          view '404'
      end



```
Thanks!

**Comment by jeremyevans** [comment]
You can also use the `not_found` plugin for this.

```ruby
plugin :not_found do
  view '404'
end
```

in your route:

```ruby
  view :single if @content
```

By default, the `not_found` plugin block is called whenever have you have a request where the response body has not been written to (route blocks that return a String write to the body), unless you have modified the status to be something other than 404.

**Reply by studiopatagon** [reply]
Works great. Thanks!
